examples/deploykey.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: gitlab.crossplane.io/v1alpha1
+kind: DeployKey
+metadata:
+  name: example-deploy-key
+spec:
+  forProvider:
+    projectIdRef:
+      name: example-project
+    title: example description
+    canPush: false
+  providerConfigRef:
+    name: gitlab-provider
+  # a reference to a Kubernetes secret to which the controller will write the private ssh key
+  writeConnectionSecretToRef:
+    name: gitlab-project-example-deploykey
+    namespace: crossplane-system
```

Comment:
I checked provider-aws and it seems there is no clear guidance (some are `default`, some are `crossplane-system`). Since this is a cluster-scoped resource, I feel that `crossplane-system` is the better place.

Reply:
But usually, as a user, you would not have access to `crossplane-system` (or at least should not use it). Since the secret is something that should be accessed by another application, it would not make sense to leave it there, would it?

Reply:
The way user consume these CRs are via crossplane recource claims (-> XRCs/compositions), which are namespaced and have a dedicated mechanism for synching secrets.

Reply:
Dan wrote a good deep dive on the topic: https://danielmangum.com/posts/crossplane-infrastructure-llvm/

Reply:
Great article, thanks for sharing! I think you are right. Lets leave it as it is.